default/python/AI/PolicyAI.py

```diff
@@ -659,6 +658,7 @@ def _adopt(self, name: str) -> None:
         if fo.issueAdoptPolicyOrder(name, category, slot):
             if name not in self._originally_adopted:
                 self._ip -= policy.adoptionCost()
+            stats.adopt_policy(name)
```

Comment:
This produces a lot of output since PolicyAI adopts and deadopts several policies temporarily.
I suggest to do the output at the end of generate_orders by comparing _adopted with _originally_adopted

Reply:
This will make things a bit more complicated.  145 items for 164 turns does not look a lot.